### PR TITLE
allow server selection based on ds assignment

### LINKED ISSUE #2582: GET /api/1.1/servers?dsId=x no longer only returns only servers associated with the ds
GET /api/1.1/servers?dsId=x and
GET /api/1.1/servers

appear to return the same result - all servers.

looks like the dsId query parameter is simply being ignored in the new Go version of the API.

https://github.com/apache/trafficcontrol/blob/master/traffic_ops/app/lib/API/Server.pm#L425

### PR DESCRIPTION
**Pull request by dangogh**
Fixes #2582

### CODE REVIEW

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2146/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2147/
Test PASSed.

**Comment by dangogh** [comment]
ah,   yes..    did not see that part of it.   Also needs to forbid examining servers for a ds outside the user's tenant.   Will mark as WIP until that's addressed..

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2150/
Test PASSed.

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2157/
Test PASSed.

**Comment by dewrich** [comment]
@dangogh Unit Tests Fails:

```BUILD| github.com/apache/trafficcontrol/traffic_ops/traffic_ops_golang/server
     | server/servers_test.go:164:54: cannot use auth.PrivLevelAdmin (type int) as type *auth.CurrentUser in argument to getServers```

**Comment by dangogh** [comment]
@dewrich fixed test and rebased

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/trafficcontrol-PR/2179/
Test PASSed.

File: traffic_ops/traffic_ops_golang/server/servers.go [hunk 1]
```diff
@@ -225,15 +264,77 @@ func getServers(params map[string]string, tx *sqlx.Tx, privLevel int) ([]tc.Serv
...
+	if len(servers) == 0 {
+		return nil, nil, tc.NoError
+	}
+	var ids []string
+	for _, s := range servers {
+		ids = append(ids, strconv.Itoa(*s.ID))
+	}
+
+	edgeIDs := strings.Join(ids, ",")
+	// TODO: include secondary parent?
```
Comment (dewrich) [inline traffic_ops/traffic_ops_golang/server/servers.go]: this TODO something to worry about?

Comment (dangogh) [reply traffic_ops/traffic_ops_golang/server/servers.go]: no -- Perl version does not refer to secondary parent at all

**Pull request merged**